src/test-helper.js

```diff
@@ -667,64 +639,68 @@ function verifyAccessDenied(doc, oldDoc, expectedAuthorization) {
   requireRole.throwWith(roleAccessDeniedError);
   requireUser.throwWith(userAccessDeniedError);
 
+  var syncFuncError = null;
   try {
     syncFunction(doc, oldDoc);
-    assert.fail('Document authorization succeeded when it was expected to fail');
   } catch (ex) {
+    syncFuncError = ex;
+  }
+
+  if (syncFuncError) {
     if (typeof(expectedAuthorization) === 'string' || expectedAuthorization instanceof Array) {
       assert.equal(
-        ex,
+        syncFuncError,
         channelAccessDeniedError,
-        'Document authorization error does not indicate channel access was denied. Actual: ' + JSON.stringify(ex));
+        'Document authorization error does not indicate channel access was denied. Actual: ' + JSON.stringify(syncFuncError));
     } else if (countAuthorizationTypes(expectedAuthorization) === 0) {
       verifyRequireAccess([ ]);
     } else if (countAuthorizationTypes(expectedAuthorization) > 1) {
       assert.equal(
-        ex.forbidden,
+        syncFuncError.forbidden,
         generalAuthFailedMessage,
-        'Document authorization error does not indicate that channel, role and user access were all denied. Actual: ' + JSON.stringify(ex));
+        'Document authorization error does not indicate that channel, role and user access were all denied. Actual: ' + JSON.stringify(syncFuncError));
     } else if (expectedAuthorization.expectedChannels) {
       assert.equal(
-        ex,
+        syncFuncError,
         channelAccessDeniedError,
-        'Document authorization error does not indicate channel access was denied. Actual: ' + JSON.stringify(ex));
+        'Document authorization error does not indicate channel access was denied. Actual: ' + JSON.stringify(syncFuncError));
     } else if (expectedAuthorization.expectedRoles) {
       assert.equal(
-        ex,
+        syncFuncError,
         roleAccessDeniedError,
-        'Document authorization error does not indicate role access was denied. Actual: ' + JSON.stringify(ex));
+        'Document authorization error does not indicate role access was denied. Actual: ' + JSON.stringify(syncFuncError));
     } else if (expectedAuthorization.expectedUsers) {
       assert.ok(
-        ex,
+        syncFuncError,
         userAccessDeniedError,
-        'Document authorization error does not indicate user access was denied. Actual: ' + JSON.stringify(ex));
+        'Document authorization error does not indicate user access was denied. Actual: ' + JSON.stringify(syncFuncError));
     }
-  }
 
-  verifyAuthorization(expectedAuthorization);
+    verifyAuthorization(expectedAuthorization);
+  } else {
+    assert.fail('Document authorization succeeded when it was expected to fail');
+  }
 }
 
 function verifyUnknownDocumentType(doc, oldDoc) {
+  var syncFuncError = null;
   try {
     syncFunction(doc, oldDoc);
-    assert.fail('Document type was successfully identified when it was expected to be unknown');
```

Comment:
If this statement executes, it throws an exception that is subsequently caught by the `catch` block and mishandled. The result is now verified outside of the try-catch to avoid the issue.